In [14]:
import os
import glob
import pandas as pd
import numpy as np

In [67]:
directory =os.getenv('HOME') + '/aiffel/aiffelthon/Data/'

In [40]:
raw = pd.read_csv('/aiffel/aiffel/aiffelthon/Data/raw_data.csv')
test = pd.read_csv('/aiffel/aiffel/aiffelthon/Data/Test_set_data')

In [44]:
raw_copy = raw.copy()
test_copy = test.copy()

In [24]:
raw

text  \
0                                            생각이 쪼금씩 바뀌더라고   
1                                  어 알겠는가 외국인들이 그래도 잘 적응하고   
2                                 아으 갑자기 예전에 맹장 뜯은 게 생각난다.   
3                    보면은 조금 품위 있게 나이 들어가야 되겠단 생각을 참 많이 해요.   
4                             약간 조금 약간 맛이 쪼끔 거시기 하긴 한데 그래도   
...                                                    ...   
1066413  뭐~ 맨날 집에서 인제 음식도 그냥 가까운 데 나가서 먹는 게 아니라 되게 시켜먹는...   
1066414                  너는 만약에 그런 시대가 오면 네가 어떻게 활동할 것 같애?   
1066415      하이브도 있고 #조석 작가 이제 조의 영역이라는 웹툰도 있고 마음의 소리도 있는데   
1066416        추석도 친척들도 사촌까지만 모이긴 했는데 이~ 다 모이니까 할 말도 많고 이~   
1066417             커피숍 가서 이렇게 먹고 하는 거가 훨씬 맛있거든 기분상으로도 그렇고   

                                                      dial reg  \
0                                            생각이 쪼금씩 바뀌드라고  jd   
1                                  어 알겄는가 외국인들이 그래도 잘 적응하고  jd   
2                                아으 갑자기 예전에 맹장 뜯은 게 생각난디야.  jd   
3                     보면은 쫌 품위 있게 나이 들어가야 되겠단 생각을 참 많이 해요.  jd   
4                              약간 쫌 약간 맛이 쪼끔 거시기 하긴 한데 그래도  jd   
...                                                    ...  ..   
1066413  뭐~ 맨날 집에서 인제 음식도 그냥 가까운 데 나가서 먹는 게 아니라 되게 시켜먹는...  jd   
1066414                  너는 만약에 그런 시대가 오면 니가 어떻게 활동할 것 같애?  gs   
1066415       하이브도 있고 조석 작가 이제 조의 영역이라는 웹툰도 있고 마음의 소리도 있는디  jd   
1066416        추석도 친척들도 사촌까지만 모이긴 했는데 이~ 다 모이난이 할 말도 많고 이~  jj   
1066417              커피숍 가서 이케 먹고 하는 거가 훨씬 맛있거든 기분상으로도 그렇고  cc   

                                                       eng  
0                       I've changed my mind a little bit.  
1          You know what? Foreigners still get used to it.  
2        All of a sudden, I think I've ripped off my ap...  
3        I have a lot of ideas about getting older in a...  
4              It's a little bit of a tastey, but still...  
...                                                    ...  
1066413  I suppose I dont just go out to the nearest pl...  
1066414  How do you think you will be active when that ...  
1066415  There is a hive a webtoon called the area of J...  
1066416  On Chuseok relatives and cousins only gathered...  
1066417  Its much better to eat at a coffee shop and it...  

[1066418 rows x 4 columns]

In [25]:
test

text  \
0                            그니까 거기 그 사진 찍는아니야 주상절리 마냥   
1                                               집에 있으면   
2                      먹고 취하잖아 그러면 취하면 한이 있잖아 사람이 왜 이게   
3                                           그렇지 그렇게하겠지   
4                나 아는사람이라고 해서 십만 원 더 지원해서 이십만 원 지원해 줬어   
..                                                 ...   
495  뭐~ 두 분 쓰시는 게 없어서 뭐~ 안정적인 건지 살 잘 얘기를 안 하시더라고요 그...   
496                   잘 치거나 이렇게 했을 때 다 그냥 이렇게 손 한 번씩 다   
497                                        아니 잘 안 맞지 넌   
498                   진짜 너무 이거 조금 조금 미안하기도 하고 다른 사람들한테   
499  바퀴벌레가 방역을 해도 기어나와 가지고 이제는 쪼끔 마음이 조금 나아지긴 했는데 처...   

                                                  dial reg  \
0                             그니까 거기 그 사진 찍는아니 주상절리 마냥  jj   
1                                               집에 이시믄  jj   
2                   먹어지믄 취하잖아 겅하*믄이 취하면 한이 있네 사람이 왜 이게  jj   
3                                            그렇지 겅하겠주게  jj   
4                 나 아는사람이라고 해서 십만 원 더 지원해서 이십만 원 지원해 줜  jj   
..                                                 ...  ..   
495  뭐~ 두 분 쓰시는 게 없어서 뭐~ 안정적인 건지 살 잘 얘기를 안 하시더라고요 그...  gs   
496                    잘 치거나 이렇게 했을 때 다 그냥 이케 손 한 번씩 다  gs   
497                                        아니 잘 안 맞지 닌  gs   
498                   진짜 너무 이거 쪼끔 쪼끔 미안하기도 하고 다른 사람들한테  gs   
499  바퀴벌레가 방역을 해도 기어나와 가지고 이제는 쪼끔 마음이 쫌 나아지긴 했는데 처음...  gs   

                                                   eng  
0    So youre not taking that picture there are you...  
1                                   If you are at home  
2    You're eating and getting drunk, and if you're...  
3                              Thats right Ill do that  
4    I ot it as I know so I applied for another 100...  
..                                                 ...  
495  Im worried about whether its stable or not bec...  
496  When you hit well or did this, you're all just...  
497                          No it doesnt fit well you  
498                           I'm so sorry about this.  
499  I fot it out even if the cockroaches crawled o...  

[500 rows x 4 columns]

In [26]:
raw.isnull().sum()

text    0
dial    0
reg     0
eng     0
dtype: int64

In [28]:
raw.isna().sum()

text    0
dial    0
reg     0
eng     0
dtype: int64

In [27]:
test.isnull().sum()

text    0
dial    0
reg     0
eng     0
dtype: int64

In [29]:
test.isna().sum()

text    0
dial    0
reg     0
eng     0
dtype: int64

In [30]:
raw.duplicated().sum()

0

In [46]:
test.duplicated().sum()

0

## train set 추출

In [54]:
train = pd.merge(raw, test, how = 'outer', indicator = True)

In [56]:
train

text  \
0                                            생각이 쪼금씩 바뀌더라고   
1                                  어 알겠는가 외국인들이 그래도 잘 적응하고   
2                                 아으 갑자기 예전에 맹장 뜯은 게 생각난다.   
3                    보면은 조금 품위 있게 나이 들어가야 되겠단 생각을 참 많이 해요.   
4                             약간 조금 약간 맛이 쪼끔 거시기 하긴 한데 그래도   
...                                                    ...   
1066413  뭐~ 맨날 집에서 인제 음식도 그냥 가까운 데 나가서 먹는 게 아니라 되게 시켜먹는...   
1066414                  너는 만약에 그런 시대가 오면 네가 어떻게 활동할 것 같애?   
1066415      하이브도 있고 #조석 작가 이제 조의 영역이라는 웹툰도 있고 마음의 소리도 있는데   
1066416        추석도 친척들도 사촌까지만 모이긴 했는데 이~ 다 모이니까 할 말도 많고 이~   
1066417             커피숍 가서 이렇게 먹고 하는 거가 훨씬 맛있거든 기분상으로도 그렇고   

                                                      dial reg  \
0                                            생각이 쪼금씩 바뀌드라고  jd   
1                                  어 알겄는가 외국인들이 그래도 잘 적응하고  jd   
2                                아으 갑자기 예전에 맹장 뜯은 게 생각난디야.  jd   
3                     보면은 쫌 품위 있게 나이 들어가야 되겠단 생각을 참 많이 해요.  jd   
4                              약간 쫌 약간 맛이 쪼끔 거시기 하긴 한데 그래도  jd   
...                                                    ...  ..   
1066413  뭐~ 맨날 집에서 인제 음식도 그냥 가까운 데 나가서 먹는 게 아니라 되게 시켜먹는...  jd   
1066414                  너는 만약에 그런 시대가 오면 니가 어떻게 활동할 것 같애?  gs   
1066415       하이브도 있고 조석 작가 이제 조의 영역이라는 웹툰도 있고 마음의 소리도 있는디  jd   
1066416        추석도 친척들도 사촌까지만 모이긴 했는데 이~ 다 모이난이 할 말도 많고 이~  jj   
1066417              커피숍 가서 이케 먹고 하는 거가 훨씬 맛있거든 기분상으로도 그렇고  cc   

                                                       eng     _merge  
0                       I've changed my mind a little bit.  left_only  
1          You know what? Foreigners still get used to it.  left_only  
2        All of a sudden, I think I've ripped off my ap...  left_only  
3        I have a lot of ideas about getting older in a...  left_only  
4              It's a little bit of a tastey, but still...  left_only  
...                                                    ...        ...  
1066413  I suppose I dont just go out to the nearest pl...  left_only  
1066414  How do you think you will be active when that ...  left_only  
1066415  There is a hive a webtoon called the area of J...  left_only  
1066416  On Chuseok relatives and cousins only gathered...  left_only  
1066417  Its much better to eat at a coffee shop and it...  left_only  

[1066418 rows x 5 columns]

In [57]:
print(train['_merge'].unique())

['left_only', 'both']
Categories (3, object): ['left_only', 'right_only', 'both']


In [58]:
print(train['_merge'].value_counts())

left_only     1065918
both              500
right_only          0
Name: _merge, dtype: int64


In [64]:
train.query('_merge == "left_only"').drop(columns=['_merge'])

text  \
0                                            생각이 쪼금씩 바뀌더라고   
1                                  어 알겠는가 외국인들이 그래도 잘 적응하고   
2                                 아으 갑자기 예전에 맹장 뜯은 게 생각난다.   
3                    보면은 조금 품위 있게 나이 들어가야 되겠단 생각을 참 많이 해요.   
4                             약간 조금 약간 맛이 쪼끔 거시기 하긴 한데 그래도   
...                                                    ...   
1066413  뭐~ 맨날 집에서 인제 음식도 그냥 가까운 데 나가서 먹는 게 아니라 되게 시켜먹는...   
1066414                  너는 만약에 그런 시대가 오면 네가 어떻게 활동할 것 같애?   
1066415      하이브도 있고 #조석 작가 이제 조의 영역이라는 웹툰도 있고 마음의 소리도 있는데   
1066416        추석도 친척들도 사촌까지만 모이긴 했는데 이~ 다 모이니까 할 말도 많고 이~   
1066417             커피숍 가서 이렇게 먹고 하는 거가 훨씬 맛있거든 기분상으로도 그렇고   

                                                      dial reg  \
0                                            생각이 쪼금씩 바뀌드라고  jd   
1                                  어 알겄는가 외국인들이 그래도 잘 적응하고  jd   
2                                아으 갑자기 예전에 맹장 뜯은 게 생각난디야.  jd   
3                     보면은 쫌 품위 있게 나이 들어가야 되겠단 생각을 참 많이 해요.  jd   
4                              약간 쫌 약간 맛이 쪼끔 거시기 하긴 한데 그래도  jd   
...                                                    ...  ..   
1066413  뭐~ 맨날 집에서 인제 음식도 그냥 가까운 데 나가서 먹는 게 아니라 되게 시켜먹는...  jd   
1066414                  너는 만약에 그런 시대가 오면 니가 어떻게 활동할 것 같애?  gs   
1066415       하이브도 있고 조석 작가 이제 조의 영역이라는 웹툰도 있고 마음의 소리도 있는디  jd   
1066416        추석도 친척들도 사촌까지만 모이긴 했는데 이~ 다 모이난이 할 말도 많고 이~  jj   
1066417              커피숍 가서 이케 먹고 하는 거가 훨씬 맛있거든 기분상으로도 그렇고  cc   

                                                       eng  
0                       I've changed my mind a little bit.  
1          You know what? Foreigners still get used to it.  
2        All of a sudden, I think I've ripped off my ap...  
3        I have a lot of ideas about getting older in a...  
4              It's a little bit of a tastey, but still...  
...                                                    ...  
1066413  I suppose I dont just go out to the nearest pl...  
1066414  How do you think you will be active when that ...  
1066415  There is a hive a webtoon called the area of J...  
1066416  On Chuseok relatives and cousins only gathered...  
1066417  Its much better to eat at a coffee shop and it...  

[1065918 rows x 4 columns]

In [65]:
Train = train.query('_merge == "left_only"').drop(columns=['_merge'])

In [66]:
Train

text  \
0                                            생각이 쪼금씩 바뀌더라고   
1                                  어 알겠는가 외국인들이 그래도 잘 적응하고   
2                                 아으 갑자기 예전에 맹장 뜯은 게 생각난다.   
3                    보면은 조금 품위 있게 나이 들어가야 되겠단 생각을 참 많이 해요.   
4                             약간 조금 약간 맛이 쪼끔 거시기 하긴 한데 그래도   
...                                                    ...   
1066413  뭐~ 맨날 집에서 인제 음식도 그냥 가까운 데 나가서 먹는 게 아니라 되게 시켜먹는...   
1066414                  너는 만약에 그런 시대가 오면 네가 어떻게 활동할 것 같애?   
1066415      하이브도 있고 #조석 작가 이제 조의 영역이라는 웹툰도 있고 마음의 소리도 있는데   
1066416        추석도 친척들도 사촌까지만 모이긴 했는데 이~ 다 모이니까 할 말도 많고 이~   
1066417             커피숍 가서 이렇게 먹고 하는 거가 훨씬 맛있거든 기분상으로도 그렇고   

                                                      dial reg  \
0                                            생각이 쪼금씩 바뀌드라고  jd   
1                                  어 알겄는가 외국인들이 그래도 잘 적응하고  jd   
2                                아으 갑자기 예전에 맹장 뜯은 게 생각난디야.  jd   
3                     보면은 쫌 품위 있게 나이 들어가야 되겠단 생각을 참 많이 해요.  jd   
4                              약간 쫌 약간 맛이 쪼끔 거시기 하긴 한데 그래도  jd   
...                                                    ...  ..   
1066413  뭐~ 맨날 집에서 인제 음식도 그냥 가까운 데 나가서 먹는 게 아니라 되게 시켜먹는...  jd   
1066414                  너는 만약에 그런 시대가 오면 니가 어떻게 활동할 것 같애?  gs   
1066415       하이브도 있고 조석 작가 이제 조의 영역이라는 웹툰도 있고 마음의 소리도 있는디  jd   
1066416        추석도 친척들도 사촌까지만 모이긴 했는데 이~ 다 모이난이 할 말도 많고 이~  jj   
1066417              커피숍 가서 이케 먹고 하는 거가 훨씬 맛있거든 기분상으로도 그렇고  cc   

                                                       eng  
0                       I've changed my mind a little bit.  
1          You know what? Foreigners still get used to it.  
2        All of a sudden, I think I've ripped off my ap...  
3        I have a lot of ideas about getting older in a...  
4              It's a little bit of a tastey, but still...  
...                                                    ...  
1066413  I suppose I dont just go out to the nearest pl...  
1066414  How do you think you will be active when that ...  
1066415  There is a hive a webtoon called the area of J...  
1066416  On Chuseok relatives and cousins only gathered...  
1066417  Its much better to eat at a coffee shop and it...  

[1065918 rows x 4 columns]

In [68]:
Train.to_csv(directory + 'Train_set_sampling.csv' , index=False)

## refrence

https://codealone.tistory.com/78

1. raw date에서 test set을 어떻게 분리해야할까 ? 라는 고민 
2. 두 개의 DataFrame에서 중복된 데이터만 분리하는 방법에 대해 구글링 
3. merge > query > drop의 방법으로 진행 

[merge]
1. 두 개의 DataFrame 을 merge 시켜준다 
2. 여기서 indicator인자를 True로 설정하여 출력시 _merge라는 열이 생성된다. 
3. _merge라는 행에는 두 개의 DataFrame에 있는 데이터의 개수와 중복되는 데이터 개수를 출력해준다. 
4. left_only 개수가 1065918 , both 개수가 500 인데 내가 원하는 데이터는 left_only


[query]
1. query함수를 사용하여 left_only만 추출 
2. query('cloumns == 원하는 태그 ') > 를 사용하여 내가 원하는 데이터만 필터링 


[drop]
1. drop = (columns == ['cloumns]) > 을 사용하여 필터링한 데이터를 제외하고 drop